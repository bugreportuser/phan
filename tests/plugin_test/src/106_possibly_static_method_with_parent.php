<?php

class PSMBase106 {
    public function examplePublic() {}

    protected function exampleProtected() {}

    private function examplePrivate() {}

    public function main() {
        $this->examplePublic();
        $this->exampleProtected();
        $this->examplePrivate();
    }
}

class PSMChild106 extends PSMBase106 {
    public function examplePublic() {}

    protected function exampleProtected() {}

    private function examplePrivate() {}

    public function main() {
        $this->examplePublic();
        $this->exampleProtected();
        $this->examplePrivate();
    }
}

(new PSMBase106())->main();
(new PSMChild106())->main();
