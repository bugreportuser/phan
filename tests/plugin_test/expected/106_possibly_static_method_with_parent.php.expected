src/106_possibly_static_method_with_parent.php:4 PhanPluginPossiblyStaticPublicMethod Public method \PSMBase106::examplePublic can be static
src/106_possibly_static_method_with_parent.php:6 PhanPluginPossiblyStaticProtectedMethod Protected method \PSMBase106::exampleProtected can be static
src/106_possibly_static_method_with_parent.php:8 PhanPluginPossiblyStaticPrivateMethod Private method \PSMBase106::examplePrivate can be static
